src/ToricVarieties/NormalToricVarieties/attributes.jl

```diff
@@ -317,6 +317,106 @@ end
+    end
+    set_attribute!(v, :coordinate_names_of_torus, coordinate_names)
+end
+export set_coordinate_names_of_torus
+
+
+@doc Markdown.doc"""
+    coordinate_names_of_torus(v::AbstractNormalToricVariety)
+
+This method returns the names of the coordinates of the torus of
+the normal toric variety `v`. The default is `x1,...,xn`.
+"""
+coordinate_names_of_torus(v::AbstractNormalToricVariety) = get_attribute!(v, :coordinate_names_of_torus, ["x$(i)" for i in 1:ambient_dim(v)])
+export coordinate_names_of_torus
+
+
+@doc Markdown.doc"""
+    coordinate_ring_of_torus(v::AbstractNormalToricVariety, R::MPolyRing)
+
+Computes the coordinate ring of the torus of the normal toric variety `v`
+in the given polynomial ring `R`.
+"""
+function coordinate_ring_of_torus(v::AbstractNormalToricVariety, R::MPolyRing)
+    if length(gens(R)) < 2 * length(coordinate_names_of_torus(v))
+        throw(ArgumentError("The given ring must have at least $(length( coordinate_names_of_torus(v))) indeterminates."))
+    end
+    relations = [gens(R)[i] * gens(R)[i+length(coordinate_names_of_torus(v))] - one(coefficient_ring(R)) for i in 1:length(coordinate_names_of_torus(v))]
+    return quo(R, ideal(relations))[1]
+end
+export coordinate_ring_of_torus
+
+
+@doc Markdown.doc"""
+    coordinate_ring_of_torus(v::AbstractNormalToricVariety)
+
+Computes the coordinate ring of the torus of the normal toric variety `v`.
+
+# Examples
+```jldoctest
+julia> p2 = projective_space(NormalToricVariety, 2);
+
+julia> set_coordinate_names_of_torus(p2, ["y1", "y2"])
+
+julia> coordinate_ring_of_torus(p2)
+Quotient of Multivariate Polynomial Ring in y1, y2, y1_, y2_ over Rational Field by ideal(y1*y1_ - 1, y2*y2_ - 1)
+```
+"""
+function coordinate_ring_of_torus(v::AbstractNormalToricVariety)
+    S, _ = PolynomialRing(coefficient_ring(v), vcat(coordinate_names_of_torus(v), [x*"_" for x in coordinate_names_of_torus(v)]), cached=false)
+    return coordinate_ring_of_torus(v, S)
+end
+export coordinate_ring_of_torus
+
+
+@doc Markdown.doc"""
+    character_to_rational_function(v::AbstractNormalToricVariety, character::Vector{fmpz}, R::MPolyRing)
+
+Computes the rational function corresponding to a character of the normal toric variety `v`.
+
+# Examples
+```jldoctest
+julia> p2 = projective_space(NormalToricVariety, 2);
+
+julia> character_to_rational_function(p2, [-1,2])
+x2^2*x1_
+```
+"""
+function character_to_rational_function(v::AbstractNormalToricVariety, character::Vector{fmpz}, R::MPolyRing)
+    if ambient_dim(v) != length(character)
+        throw(ArgumentError("A character consist of as many integers as the ambient dimension of the variety."))
+    end
+    generators = gens(coordinate_ring_of_torus(v,R))
+    rational_function = one(coefficient_ring(R))
+    for i in 1:length(character)
+        if character[i] < 0
+            rational_function = rational_function * generators[i+ambient_dim(v)]^(-1*character[i])
```

Comment:
```suggestion
            rational_function *= generators[i+ambient_dim(v)]^(-1*character[i])
```
And similarly elsewhere

Reply:
Thx. Adjsuted.